projects/plugins/jetpack/extensions/blocks/seo/noindex-panel.js

```diff
@@ -0,0 +1,27 @@
+import { CheckboxControl } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+
+const SeoNoindexPanel = () => {
+	const { postMeta } = useSelect( select => {
+		return {
+			postMeta: select( 'core/editor' ).getEditedPostAttribute( 'meta' ),
+		};
+	} );
+	const { editPost } = useDispatch( 'core/editor', [ postMeta.jetpack_seo_noindex ] );
```

Comment:
It might need improvement to support `boolean` values but shouldn't be that difficult. E.g. we can improve the function params to:
`export const withSeoHelper = ( attrName, defaultValue = '' ) =>`

Reply:
Good suggestions, thanks! fb927c6 and 7a13555